homeassistant/components/crownstone/entry_manager.py

```diff
@@ -0,0 +1,205 @@
+"""Code to set up all communications with Crownstones."""
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any, cast
+
+from crownstone_cloud import CrownstoneCloud
+from crownstone_cloud.exceptions import (
+    CrownstoneAuthenticationError,
+    CrownstoneUnknownError,
+)
+from crownstone_sse import CrownstoneSSEAsync
+from crownstone_uart import CrownstoneUart, UartEventBus
+
+from homeassistant.components import persistent_notification
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    CONF_EMAIL,
+    CONF_PASSWORD,
+    CONF_UNIQUE_ID,
+    EVENT_HOMEASSISTANT_STOP,
+)
+from homeassistant.core import Event, HomeAssistant, callback
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+
+from .const import CONF_USB_PATH, CONF_USE_CROWNSTONE_USB, DOMAIN, PLATFORMS, SSE, UART
+from .helpers import get_port
+from .listeners import create_data_listeners
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class CrownstoneEntryManager:
+    """Manage a Crownstone config entry."""
+
+    def __init__(self, hass: HomeAssistant, config_entry: ConfigEntry) -> None:
+        """Initialize the hub."""
+        self.hass = hass
+        self.config_entry = config_entry
+        self.uart = cast(CrownstoneUart, None)
+        self.cloud = cast(CrownstoneCloud, None)
+        self.sse = cast(CrownstoneSSEAsync, None)
```

Comment:
I usually put fields that are initialized later in `__init__()` so it's clear what fields that class actually has. Just `self.uart = None` doesn't go well with mypy however, so this is what i came up with. In case of uart this fields stays `None` in case a USB fails to setup, and in `light.py`  it is compared with a `None` to see if the cloud or a USB should be used. 

What would you recommend?

Reply:
Initialize them as class attributes:

```py
   uart: CrownstoneUart
   cloud: CrownstoneCloud
   sse: CrownstoneSSEAsync

   def __init__...
```